src/explore-education-statistics-admin/src/pages/publication/components/__tests__/ReleaseContributorPermissions.test.tsx

```diff
@@ -0,0 +1,156 @@
+    id: 'release-1',
+    title: 'Release 1',
+  } as Release;
+
+  const testReleaseContributors: ManageAccessPageContributor[] = [
+    {
+      userId: 'user-1',
+      userFullName: 'User Name 1',
+      userEmail: 'user1@test.com',
+      releaseRoleId: 'release-role-1',
+    },
+    {
+      userId: 'user-2',
+      userFullName: 'User Name 2',
+      userEmail: 'user2@test.com',
+      releaseRoleId: undefined,
+    },
+    {
+      userId: 'user-3',
+      userFullName: 'User Name 3',
+      userEmail: 'user3@test.com',
+      releaseRoleId: 'release-role-3',
+    },
+  ];
+
+  test('renders a message when there are no release contributors', async () => {
+    releasePermissionService.getPublicationReleaseContributors.mockResolvedValue(
+      [],
+    );
+
+    const handleUserRemoval = jest.fn();
+
+    render(
+      <ReleaseContributorsPermissions
+        contributors={[]}
+        handleUserRemoval={handleUserRemoval}
+      />,
+    );
+
+    await waitFor(() => {
+      expect(screen.getByText('Warning')).toBeInTheDocument();
+
+      expect(
+        screen.getByTestId('releaseContributors-warning').textContent,
+      ).toContain(
+        'There are currently no team members associated to this publication.',
+      );
+    });
+  });
+
+  test('renders the contributors table correctly', async () => {
+    releasePermissionService.getPublicationReleaseContributors.mockResolvedValue(
+      testReleaseContributors,
+    );
+
+    const handleUserRemoval = (userId: string) => {};
+
+    render(
+      <ReleaseContributorsPermissions
+        contributors={testReleaseContributors}
+        handleUserRemoval={handleUserRemoval}
+      />,
+    );
+
+    await waitFor(() => {
+      expect(screen.getByRole('table')).toBeInTheDocument();
+    });
+
+    const rows = screen.getAllByRole('row');
+    expect(rows.length).toBe(3);
+
+    expect(within(rows[0]).getByText('User Name 1')).toBeInTheDocument();
+    expect(
+      within(rows[0]).getByRole('button', { name: 'Remove user' }),
+    ).toBeInTheDocument();
+
+    expect(within(rows[1]).getByText('User Name 2')).toBeInTheDocument();
+    expect(
+      within(rows[1]).getByRole('button', { name: 'Remove user' }),
+    ).toBeInTheDocument();
+
+    expect(within(rows[2]).getByText('User Name 3')).toBeInTheDocument();
+    expect(
+      within(rows[2]).getByRole('button', { name: 'Remove user' }),
+    ).toBeInTheDocument();
+  });
+
+  test('remove user', async () => {
+    releasePermissionService.getPublicationReleaseContributors.mockResolvedValue(
+      testReleaseContributors,
+    );
+
+    const handleUserRemoval = jest.fn();
+
+    render(
+      <ReleaseContributorsPermissions
+        contributors={testReleaseContributors}
+        handleUserRemoval={handleUserRemoval}
+      />,
+    );
+
+    await waitFor(() => {
+      expect(screen.getByRole('table')).toBeInTheDocument();
+    });
+    const rows = screen.getAllByRole('row');
+
+    fireEvent.click(
```

Comment:
We should use `userEvent` from `@testing-library/user-event` instead. Whilst it doesn't make a huge difference here, it should be our go-to for any user interactions.